torch/_dynamo/symbolic_convert.py

```diff
@@ -2641,33 +2641,46 @@ def GET_YIELD_FROM_ITER(self, inst):
             self.pop()
             res = BuiltinVariable(iter).call_function(self, [tos], {})
             self.push(res)
-        return self.YIELD_FROM(inst)
 
     def YIELD_FROM(self, inst):
-        while True:
-            tos = self.stack[-1].realize()
-            if isinstance(tos, ConstantVariable) and tos.value is None:
-                self.pop()
-                return
-            try:
-                val = tos.next_variable(self)
-                self.push(val)
-                # TODO(voz): Unclear if we need the push None in YIELD_VALUE?
-                self.YIELD_VALUE(inst)
-                self.pop()
-                self.push(tos)
-            except (StopIteration, exc.UserStopIteration):
-                # TODO(jansel): do we need a self.pop() here?
-                return
+        assert len(self.stack) >= 2
+        val = self.pop()
+        tos = self.stack[-1].realize()
+        if not (isinstance(val, ConstantVariable) and val.value is None):
+            # invoke send
+            # Unreachable code - if you hit this, you are implementing generator support and have
+            # lifted the `unimplemented("generator")` in frame conversion. This codepath handles
+            # subgenerator and lines up with this line in Python 3.10
+            # https://github.com/python/cpython/blob/3.10/Python/ceval.c#L2599
+            unimplemented("Unreachable sub-generator code")
+
+        try:
+            val = tos.next_variable(self)
+        except (StopIteration, exc.UserStopIteration) as ex:
+            self.pop()
+            self.push(ConstantVariable.create(ex.value))
```

Comment:
I think this will fail if the user code raises a StopIteration:
```
>>> torch._dynamo.exc.UserStopIteration().value
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: 'UserStopIteration' object has no attribute 'value'
```

Reply:
I add the `value` attribute to the class `UserStopIteration` in this PR.

Or we can ignore the value of the exception and push `ConstantVariable.create(None)`.

> if the user code raises a StopIteration

Can you give some hints on how can we test this? `StopIteration` will convert to `RuntimeError`.

```python
In [1]: def gen():
   ...:     yield 1
   ...:     yield 2
   ...:     raise StopIteration(3)
   ...:     

In [2]: list(gen())
StopIteration: 3

The above exception was the direct cause of the following exception:

RuntimeError: generator raised StopIteration
```

Reply:
You can test this with something like:
```
class MyIter:
  def __iter__(self):
     return self
  def __next__(self):
     raise StopIteration
```

When we trace through `__next__` we transform `StopIteration` to `UserStopIteration`.